test/template/www/js/scenarioSyncRestart30.js

```diff
@@ -0,0 +1,47 @@
+var app = {
+    // Application Constructor
+    initialize: function () {
+        this.bindEvents();
+    },
+    bindEvents: function () {
+        document.addEventListener('deviceready', this.onDeviceReady, false);
+    },
+    onDeviceReady: function () {
+        app.receivedDeviceReady();
+    },
+    // Update DOM on a Received Event
+    receivedDeviceReady: function () {
+        document.getElementById("deviceready").innerText = "Device is ready (scenario - sync)";
+        console.log('Received Event: deviceready');
+        /* invoke sync with no UI options */
+        window.codePush.sync(
+            function (status) {
+                // only output result statuses
+                switch(status) {
+                    case SyncStatus.UP_TO_DATE:
+                    case SyncStatus.UPDATE_INSTALLED:
+                    case SyncStatus.UPDATE_IGNORED:
+                    case SyncStatus.ERROR:
+                    case SyncStatus.IN_PROGRESS:
+                        app.sendTestMessage("SYNC_STATUS", [status]);
+                    default:
+                        break;
+                }
+            },
+            {
+                installMode: InstallMode.ON_NEXT_RESTART,
+                minimumBackgroundDuration: 30
```

Comment:
Totally optional, but I would go for smaller values, to avoid unnecessary increases in test time. It will save you time in the long run. I would use values like (5s, 10s, 15s) instead of (10s, 30s, 40s).